### PR TITLE
Enable TE in test_jit.py

### PR DESCRIPTION
**Pull request by Krovatkin**
Enable TE in test_jit.py and adjust/fix tests accordingly.

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -687,7 +683,7 @@ def test_input(func, input, result):
             self.assertEqual(func(input, profile_and_replay=True), result)
             gre = func.graph_for(input)
             if GRAPH_EXECUTOR == ProfilingMode.PROFILING:
-                FileCheck().check("prim::Constant").check_next("prim::BailoutTemplate").run(gre)
```
Comment (eellison) [inline test/test_jit.py:690]: We should just skip this test in profiling executor, it's not doing anything.

File: test/test_jit.py [hunk 2]
```diff
@@ -5247,7 +5241,7 @@ def func(alpha, beta, x, y):
         # NOTE: cannot optimize yet because broadcasts are not inserted before the fuser runs
         self.checkScript(func, [alpha, beta, x, y], optimize=False)
 
-    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING, "skip if profiling isn't enabled")
+    @unittest.skip("bailouts are being deprecated")
```
Comment (eellison) [inline test/test_jit.py:5244]: At some point we should just delete these tests, but it probably makes sense to delete it in the same PR that rermoves Bailouts, so we get a clean git history...

**Review by ZolotukhinM — APPROVED** [review]
Looks good to me!

File: test/test_jit.py [hunk 3]
```diff
@@ -10517,19 +10511,25 @@ def test_rand():
...
         def randint():
             return torch.randint(0, 5, [1, 2])
 
-        out = randint(profile_and_replay=True)
-        self.assertEqual(out.dtype, torch.double)
         # although the type should be int here, testing that the runtime dtype
         # and shape analysis dtype is the same.
-        FileCheck().check("Double(1:2, 2:1, requires_grad=0, device=cpu)") \
-                   .check_not("Float(1:2, 2:1, requires_grad=0, device=cpu)").run(randint.graph_for())
+        with enable_profiling_mode_for_profiling_tests():
```
Comment (eellison) [inline test/test_jit.py:10526]: In the future, tests that are only run with the profiling executor should be in a separate file

**Review by eellison — APPROVED** [review]
LGTM

**Pull request closed without merging**